### PR TITLE
[CdapIO] Complete examples for CDAP Salesforce plugins

### PR DESCRIPTION
**Pull request by Amar3tto**
Added examples for the following CDAP Salesforce plugins:

- Batch Source
- Batch Sink
- Streaming Source

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Amar3tto** [comment]
Run Java PreCommit

**Comment by Amar3tto** [comment]
Run Java_Pulsar_IO_Direct PreCommit

**Comment by Amar3tto** [comment]
Run Java_Debezium_IO_Direct PreCommit

File: examples/java/cdap/salesforce/src/main/java/org/apache/beam/examples/complete/cdap/options/BaseCdapOptions.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import io.cdap.plugin.common.Constants;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.Validation;
+
+/**
+ * The {@link BaseCdapOptions} interface provides the custom execution options passed by the
+ * executor at the command-line for examples with Cdap plugins.
+ */
+public interface BaseCdapOptions extends PipelineOptions {
```
Comment (Lizzfox) [inline examples/java/cdap/salesforce/src/main/java/org/apache/beam/examples/complete/cdap/options/BaseCdapOptions.java:29]: can we move this class to the cdap package level?

**Comment by Lizzfox** [comment]
R: @johnjcasey 
R: @aromanenko-dev 

We divided our [CDAP examples PR](https://github.com/apache/beam/pull/23501) by example (service). This one is for the Salesforce examples. 
Could you please take the time to review it?

Thanks in advance for your attention to it!

**Comment by Amar3tto** [comment]
Run Java_JDBC_IO_Direct PreCommit

**Comment by Amar3tto** [comment]
Run Java_Pulsar_IO_Direct PreCommit

**Comment by Amar3tto** [comment]
Run Java_Pulsar_IO_Direct PreCommit

**Comment by Amar3tto** [comment]
Run SQL_Java11 PreCommit

**Comment by Amar3tto** [comment]
Run Website PreCommit

**Comment by Lizzfox** [comment]
@johnjcasey 
We would really appreciate it if you would take the time to review the PR with the Salesforce example.
Thank you for your time and consideration!

**Review by johnjcasey — APPROVED** [review]
This looks good, but there are some merge conflicts to be resolved

**Review by Lizzfox — APPROVED** [review]

**Comment by Lizzfox** [comment]
@johnjcasey 
all conflicts fixed
Thank you!

**Comment by Lizzfox** [comment]
@johnjcasey 
friendly reminder that all conflicts that appeared after the [example with Hubspot](https://github.com/apache/beam/pull/24568) merged have been fixed and this PR is ready to merge

Thank you!

**Comment by Abacn** [comment]
Thanks @Amar3tto and @Lizzfox!

**Comment by Abacn** [comment]
well, the newly triggered python tests are irrelevant. Merge for now

**Pull request merged**